src/measure_theory/lp_space.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2020 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Rémy Degenne.
+-/
+import measure_theory.l1_space
+import analysis.special_functions.pow
+
+/-!
+# ℒp space
+
+This file describes properties of measurable functions with finite seminorm
+`(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `p:ℝ` with `1 ≤ p`.
+
+## Main definitions
+
+* `mem_ℒp f p μ` : the function `f` has finite p-seminorm for measure `μ`, for `p:ℝ` such that
+                  `hp1 : 1 ≤ p`,
+
+## Notation
+
+* `snorm f p μ` : `(∫⁻ a, (nnnorm (f a))^(p : ℝ) ∂μ) ^ (1/p)` for `f : α → β`
+
+-/
+
+open measure_theory
+
+noncomputable theory
+
+namespace ℒp_space
+
+variables {α β γ: Type*} [measurable_space α] {μ : measure α}
+  [measurable_space β] [normed_group β]
+  [normed_group γ]
+  {p : ℝ}
+
+section ℒp_space_definition
+
+/-- The property that f belongs to ℒp for measure μ and real p -/
+def mem_ℒp (f : α → β) (p : ℝ) (μ : measure α) : Prop :=
+measurable f ∧ ∫⁻ a, (nnnorm (f a)) ^ p ∂μ < ⊤
+
+/-- seminorm on ℒp -/
+def snorm (f : α → γ) (p : ℝ) (μ : measure α) : ennreal := (∫⁻ a, (nnnorm (f a))^p ∂μ) ^ (1/p)
+
+lemma lintegral_rpow_nnnorm_eq_rpow_snorm {f : α → γ} (hp0_lt : 0 < p) :
+  ∫⁻ a, (nnnorm (f a)) ^ p ∂μ = (snorm f p μ) ^ p :=
+begin
+  unfold snorm,
+  rw [←ennreal.rpow_mul, one_div, inv_mul_cancel, ennreal.rpow_one],
+  symmetry, exact ne_of_lt hp0_lt,
+end
+
+end ℒp_space_definition
+
+lemma mem_ℒp_one_iff_integrable : ∀ f : α → β, mem_ℒp f 1 μ ↔ integrable f μ :=
+begin
+  intro f,
+  unfold integrable, unfold has_finite_integral, unfold mem_ℒp,
+  simp only [ennreal.rpow_one, nnreal.coe_one],
+end
+
+section top
+
+lemma snorm_lt_top {f : α → β} (hp0 : 0 ≤ p) (hfp : mem_ℒp f p μ) : snorm f p μ < ⊤ :=
```

Comment:
It is not clear to me how I would do that. What namespace do they need to be in for that dot notation to work? Do you have a link to documentation about it that I could read?

Reply:
See the end of [3.3.1](https://leanprover.github.io/theorem_proving_in_lean/propositions_and_proofs.html#conjunction) and [9.1](https://leanprover.github.io/theorem_proving_in_lean/structures_and_records.html) in TPiL. The discussion in [this thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/dot.20notation.20confusion/near/157341136) was also helpful when I was learning about it.

Reply:
Here you just need to replace the name of the lemma with `mem_ℒp.snorm_lt_top`. Then in the proof of the next lemma you can write `hfp.snorm_lt_top hp0` instead of `snorm_lt_top hp0 hfp`. It may look like a trivial change, but in practice this is extremely helpful to make proofs both more readable and more streamlined.

Reply:
thanks for the pointers and explanation!